src/main/cpp/hardware/Hardware.cpp

```diff
@@ -80,12 +81,31 @@ bool SetupSwerveDriveInterface(
    ArmHardwareInterface arm_interface = {actuators->arm_lift_motor.get(),
                                          actuators->arm_up_motor.get(),
                                          actuators->arm_extend_motor.get(),
-                                         sensors->arm_lift_encoder.get()
+                                         sensors->arm_lift_encoder.get(),
+                                         sensors->arm_duty_cycle_encoder.get(),
+                                         sensors->arm_extend_encoder.get(),
+                                         sensors->extend_limit_switch.get()
                                          };
     interface = std::make_shared<ArmHardwareInterface>(arm_interface);
 
     return true;
+}
 
+bool SetupClawInterface(std::unique_ptr<Hardware> &hardware, std::shared_ptr<ClawHardwareInterface> &interface) {
+    //Get actuators interface for arm
+    std::unique_ptr<Actuators> &actuators = hardware->actuators;
+    std::unique_ptr<Sensors> &sensors = hardware->sensors;
+    
+   //make sure the actuators actually exist 
+   OKC_CHECK(actuators->claw_motor != nullptr);
+   
+   ClawHardwareInterface claw_interface = {
+    actuators->claw_motor.get(),
+    sensors->claw_encoder.get()
```

Comment:
We should probably OKC_CHECK the claw_encoder as well

Reply:
:+1+